x-pack/plugins/security_solution/server/lib/detection_engine/migrations/create_migration_index.ts

```diff
@@ -38,6 +38,7 @@ export const createMigrationIndex = async ({
       settings: {
         index: {
           lifecycle: {
+            // @ts-expect-error typings don't contain the property yet
```

Comment:
My IDE might just be being a bit slow, but not sure the `@ts-expect-error` is needed. Seemed fine without it. We've been trying to avoid using it in our code as it's resulted in some bugs.

Reply:
`@ts-expect-error` will fail the ES/TS lint validation if it's incorrectly used on a line not having any TS errors, so it's likely just your IDE not having fully reloaded the types yet (did you `kbn bootstrap`?)